src/linear_algebra/matrix/reindex.lean

```diff
@@ -52,14 +57,51 @@ rfl
   reindex_linear_equiv (equiv.refl m) (equiv.refl n) = linear_equiv.refl R _ :=
 linear_equiv.ext $ λ _, rfl
 
-lemma reindex_linear_equiv_mul {o o' : Type*} [fintype o] [fintype o'] [semiring R]
+lemma reindex_linear_equiv_trans [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') : (reindex_linear_equiv e₁ e₂).trans (reindex_linear_equiv e₁' e₂') =
+   @reindex_linear_equiv _ _ _ _ _ _ _ _ R _ (e₁.trans e₁') (e₂.trans e₂') :=
+by { ext, refl }
+
+lemma reindex_linear_equiv_comp [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') : (@reindex_linear_equiv _ _ _ _ _ _ _ _ R _ e₁' e₂') ∘
+   (@reindex_linear_equiv _ _ _ _ _ _ _ _ R _ e₁ e₂)
+  = (reindex_linear_equiv (e₁.trans e₁') (e₂.trans e₂')) :=
+by { rw [← reindex_linear_equiv_trans], refl }
+
+lemma reindex_linear_equiv_comp_apply [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') (M : matrix m n R) :
+  (reindex_linear_equiv e₁' e₂') (reindex_linear_equiv e₁ e₂ M) =
+    reindex_linear_equiv (e₁.trans e₁') (e₂.trans e₂') M :=
+by rw [← reindex_linear_equiv_comp e₁ e₂ e₁' e₂']
+
+@[simp] lemma reindex_linear_equiv_one [semiring R] [decidable_eq m] [decidable_eq m']
+  (e : m ≃ m') : (reindex_linear_equiv e e (1 : matrix m m R)) = 1 := by simp
```

Comment:
```suggestion
lemma reindex_linear_equiv_one [semiring R] [decidable_eq m] [decidable_eq m']
  (e : m ≃ m') : (reindex_linear_equiv e e (1 : matrix m m R)) = 1 :=
minor_one e e.injective
```

Reply:
I will certainly commit this suggestion, but can I ask for an explanation both about not marking it as `@[simp]` and about why is the `minor_one_equiv e` proof better than the `by simp` one?